File: Python/0075.py
# th logic is to count total number of 0's, 1's and 2's int the list and starting from beginning first append total 0's then total 1's and then 2's
class Solution:
    def sortColors(self, li: List[int]) -> None: 
        count1 = 0
        count2 = 0
        count3 = 0
        for i in range(len(li)): 
            if li[i] == 0: 
                count1 += 1

            elif li[i] == 1: 
                count2 += 1
            else:
                count3 += 1


        i = 0
        while (count1 > 0): 
            li[i] = 0
            i+=1
            count1 -= 1
        while (count2 > 0): 
            li[i] = 1
            i+=1
            count2 -= 1
        while (count3 > 0): 
            li[i] = 2
            i+=1
            count3 -= 1

        print(li) 

File: Java/0075.java
import java.util.*;
class Solution {
    public void sortColors(int[] array) 
    {
		HashMap<Integer, Integer> map = new HashMap<>();
		map.put(0,0);
		map.put(1,0);
		map.put(2,0);
		
		for(int i=0; i<array.length; i++)
		{
			int number = array[i];
			int count = map.get(number);
			count++;
			map.replace(number, count);
		}
		
		//Sort array
		int i = 0, arrayIndex = 0;
		int count = map.get(0);
		for(i=0; i<count; i++)
		{
			array[arrayIndex] = 0;
			arrayIndex++;
		}
		count = map.get(1);
		for(i=0; i<count; i++)
		{
			array[arrayIndex] = 1;
			arrayIndex++;
		}
		count = map.get(2);
		for(i=0; i<count; i++)
		{
			array[arrayIndex] = 2;
			arrayIndex++;
		}
    }
}